storage_controller/src/scheduler.rs

```diff
@@ -83,10 +90,56 @@ pub(crate) struct NodeAttachmentSchedulingScore {
     node_id: NodeId,
 }
 
+impl Ord for NodeAttachmentSchedulingScore {
```

Comment:
nit: it was kind of nice that we didn't have to implement `Ord` explicitly before -- could we avoid this by declaring `az_match` attribute as a type that has the ordering we want, rather than writing out the Ord for the whole score object?

Reply:
https://github.com/neondatabase/neon/pull/9083/commits/8d51003b88342dde761fdbaea06e481071e53f63